python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -542,42 +584,339 @@ def asset_key_for_input(self, input_name: str) -> AssetKey:
 
     @public
     def asset_partition_key_for_output(self, output_name: str = "result") -> str:
-        """Returns the asset partition key for the given output. Defaults to "result", which is the
-        name of the default output.
+        """Returns the asset partition key for the given output.
+
+        Args:
+            output_name (str): The name of the output produced by the asset to get the partition key for.
```

Comment:
Still working on how to best communicate what `output_name` is and when you need to provide it yourself

Reply:
How about. "The name of the the op output associated with the asset key (as determined by AssetOut) to get the partition key for."